### PR TITLE
Module4

### PR DESCRIPTION
**Pull request by SerhiiSmilianets**
Dialog and MovieForm components files created

### CODE REVIEW

**Comment by SerhiiSmilianets** [comment]
Added also button that will open the dialog and move it to component, so I will use only one import. Additional props for styling will be added or the functionality will be updated

Screenshot 2024-03-04 132154
Screenshot 2024-03-04 132205
Screenshot 2024-03-04 132214

File: src/components/Dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+const Dialog: React.FC<DialogProps> = ({title, onClose, children}) => {
+    const handleOverlayClick = (event: MouseEvent<HTMLDivElement>) => {
+        if (typeof onClose === "function") {
+            onClose()
+        }
+    }
+
+    const handleButtonClick = (event: MouseEvent<HTMLButtonElement>) => {
```
Comment (MykhailoSheliahov) [inline src/components/Dialog.tsx]:
```suggestion
    const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
```

File: src/components/Dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React, { ReactNode, MouseEvent } from 'react';
+import '../components-styles/Dialog.scss';
+
+interface DialogProps {
+  title: string;
+  onClose?: () => void;
```
Comment (MykhailoSheliahov) [inline src/components/Dialog.tsx]: Lets use one pattern for functions naming - **onEvent** or **handleEvent**

File: src/components/DialogOpenButton.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+interface MovieData {
+  title: string;
+  release_date: string;
+  poster_path: string;
+  vote_average: number;
+  genres: string[];
+  runtime: number;
+  overview: string;
+}
```
Comment (MykhailoSheliahov) [inline src/components/DialogOpenButton.tsx:14]: As it is using in a lot of places, lets move this interface  to types.ts file and import where it is needed

File: src/components/MovieForm.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import React, { FormEvent } from 'react';
+import '../components-styles/MovieForm.scss';
+import {GenreList} from '../constants'
```
Comment (MykhailoSheliahov) [inline src/components/MovieForm.tsx]: For const we should use upper case
```suggestion
import {GENRE_LIST} from '../constants'
```

File: src/components/Dialog.tsx (hunk 2 above)
Comment (MykhailoSheliahov) [inline src/components/Dialog.tsx]: Should it be required?

File: src/components/Dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  children?: ReactNode;
```
Comment (MykhailoSheliahov) [inline src/components/Dialog.tsx]: Should it be required?

**Review by MykhailoSheliahov — CHANGES_REQUESTED** [review]

File: src/components/DialogOpenButton.tsx (hunk 3 above)
Comment (SerhiiSmilianets) [reply src/components/DialogOpenButton.tsx:14]: @MykhailoSheliahov I will do this in module 5, cause will add some more tsx files and will create common file for them. Need some more experience in ts to plan in this case

**Review by MykhailoSheliahov — APPROVED** [review]

**Pull request merged**